### ISSUE TITLE
Error when portforwarding with pymongo 

### ISSUE THREAD

**Issue by GeorgeVentura** [kr8s-org/kr8s#298]
### Which project are you reporting a bug for?

kr8s

### What happened?

I'm getting this error when I port forward a pod running mongodb, connect to it using `pymongo`, run a query, on `pf.stop()` 

```python
from kr8s.objects import Pod
import pymongo

mongo_uri = 'mongodb://root:password@127.0.0.1:27017'

pod = Pod.get('mongo-mongodb-56fdf98bbd-6kpgc', namespace='default')

pf = pod.portforward(27017, 27017)
pf.start()

conn = pymongo.MongoClient(mongo_uri)
db = conn['database']
docs = db['collections'].find_one()
conn.close()

pf.stop()
```

```
Task exception was never retrieved
future: <Task finished name='Task-18' coro=<PortForward._sync_sockets() done, defined at /usr/local/lib/python3.11/site-packages/kr8s/_portforward.py:215> exception=RuntimeError('cannot schedule new futures after shutdown')>
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/kr8s/_portforward.py", line 218, in _sync_sockets
    async with self._connect_websocket() as ws:
  File "/usr/local/lib/python3.11/contextlib.py", line 211, in __aexit__
    await anext(self.gen)
  File "/usr/local/lib/python3.11/site-packages/kr8s/_portforward.py", line 197, in _connect_websocket
    async with self.pod.api.open_websocket(
  File "/usr/local/lib/python3.11/contextlib.py", line 204, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/kr8s/_api.py", line 194, in open_websocket
    async with session.ws_connect(**kwargs) as response:
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client.py", line 1194, in __aenter__
    self._resp = await self._coro
                 ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client.py", line 832, in _ws_connect
    resp = await self.request(
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client.py", line 578, in _request
    conn = await self._connector.connect(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 544, in connect
    proto = await self._create_connection(req, traces, timeout)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 911, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 1173, in _create_direct_connection
    hosts = await asyncio.shield(host_resolved)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 884, in _resolve_host
    addrs = await self._resolver.resolve(host, port, family=self._family)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/resolver.py", line 33, in resolve
    infos = await self._loop.getaddrinfo(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 864, in getaddrinfo
    return await self.run_in_executor(
                 ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 826, in run_in_executor
    executor.submit(func, *args), loop=self)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 167, in submit
    raise RuntimeError('cannot schedule new futures after shutdown')
RuntimeError: cannot schedule new futures after shutdown
Task exception was never retrieved
future: <Task finished name='Task-11' coro=<PortForward._sync_sockets() done, defined at /usr/local/lib/python3.11/site-packages/kr8s/_portforward.py:215> exception=RuntimeError('cannot schedule new futures after shutdown')>
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/kr8s/_portforward.py", line 218, in _sync_sockets
    async with self._connect_websocket() as ws:
  File "/usr/local/lib/python3.11/contextlib.py", line 211, in __aexit__
    await anext(self.gen)
  File "/usr/local/lib/python3.11/site-packages/kr8s/_portforward.py", line 197, in _connect_websocket
    async with self.pod.api.open_websocket(
  File "/usr/local/lib/python3.11/contextlib.py", line 204, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/kr8s/_api.py", line 194, in open_websocket
    async with session.ws_connect(**kwargs) as response:
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client.py", line 1194, in __aenter__
    self._resp = await self._coro
                 ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client.py", line 832, in _ws_connect
    resp = await self.request(
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client.py", line 578, in _request
    conn = await self._connector.connect(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 544, in connect
    proto = await self._create_connection(req, traces, timeout)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 911, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 1173, in _create_direct_connection
    hosts = await asyncio.shield(host_resolved)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 884, in _resolve_host
    addrs = await self._resolver.resolve(host, port, family=self._family)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/resolver.py", line 33, in resolve
    infos = await self._loop.getaddrinfo(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 864, in getaddrinfo
    return await self.run_in_executor(
                 ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 826, in run_in_executor
    executor.submit(func, *args), loop=self)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 167, in submit
    raise RuntimeError('cannot schedule new futures after shutdown')
RuntimeError: cannot schedule new futures after shutdown
Task exception was never retrieved
future: <Task finished name='Task-19' coro=<PortForward._sync_sockets() done, defined at /usr/local/lib/python3.11/site-packages/kr8s/_portforward.py:215> exception=RuntimeError('cannot schedule new futures after shutdown')>
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/kr8s/_portforward.py", line 218, in _sync_sockets
    async with self._connect_websocket() as ws:
  File "/usr/local/lib/python3.11/contextlib.py", line 211, in __aexit__
    await anext(self.gen)
  File "/usr/local/lib/python3.11/site-packages/kr8s/_portforward.py", line 197, in _connect_websocket
    async with self.pod.api.open_websocket(
  File "/usr/local/lib/python3.11/contextlib.py", line 204, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/kr8s/_api.py", line 194, in open_websocket
    async with session.ws_connect(**kwargs) as response:
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client.py", line 1194, in __aenter__
    self._resp = await self._coro
                 ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client.py", line 832, in _ws_connect
    resp = await self.request(
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client.py", line 578, in _request
    conn = await self._connector.connect(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 544, in connect
    proto = await self._create_connection(req, traces, timeout)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 911, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 1173, in _create_direct_connection
    hosts = await asyncio.shield(host_resolved)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 884, in _resolve_host
    addrs = await self._resolver.resolve(host, port, family=self._family)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/resolver.py", line 33, in resolve
    infos = await self._loop.getaddrinfo(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 864, in getaddrinfo
    return await self.run_in_executor(
                 ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 826, in run_in_executor
    executor.submit(func, *args), loop=self)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 167, in submit
    raise RuntimeError('cannot schedule new futures after shutdown')
RuntimeError: cannot schedule new futures after shutdown
```

### Anything else?

kr8s: 0.13.0
python: 3.11

**Comment by jacobtomlinson** [comment]
Thanks for raising this. Just to check does the `pymongo` query succeed?

It would also be really helpful if you could provide an example spec for the Mongo Pod you created so I can reproduce locally.

What happens if you use the context manager instead? I'm curious if this is a bug in the stop method or port forwarding as a whole.

```python
from kr8s.objects import Pod
import pymongo

mongo_uri = 'mongodb://root:password@127.0.0.1:27017'

pod = Pod.get('mongo-mongodb-56fdf98bbd-6kpgc', namespace='default')

with pod.portforward(27017, 27017) as pf:
    conn = pymongo.MongoClient(mongo_uri)
    db = conn['database']
    docs = db['collections'].find_one()
    conn.close()
```

**Comment by jacobtomlinson** [comment]
Thanks @GeorgeVentura. Any chance of providing a complete MRE with a bit of code to create/delete the mongo Pod too? It'll be much easier for me to dig into this if there is something I can just copy/paste to reproduce the issue.

**Comment by GeorgeVentura** [comment]
```python
from kr8s.objects import Pod
import pymongo
import logging
import time

pod = Pod(
    {
  "apiVersion": "v1",
  "kind": "Pod",
  "metadata": {
    "name": "mongo-pod",
    "namespace": "default"
  },
  "spec": {
    "containers": [
      {
        "name": "mongo-container",
        "image": "docker.io/bitnami/mongodb:6.0.13",
        "ports": [
          {
            "containerPort": 27017
          }
        ]
      }
    ]
  }
}
)

pod.create()

while not pod.ready():
    print("Waiting for pod to be ready.")
    time.sleep(1)

pf = pod.portforward(27017, 27017)
pf.start()

mongo_uri = 'mongodb://127.0.0.1:27017'

conn = pymongo.MongoClient(mongo_uri)
db = conn['admin']
docs = db['system.version'].find_one()
print(docs)
conn.close()

pf.stop()

pod.delete()
```

**Comment by jacobtomlinson** [comment]
Fixed by #317
